marimo/_plugins/ui/_impl/chart.py

```diff
@@ -0,0 +1,258 @@
+
+    Use `mo.ui.chart` to make Altair charts reactive: select chart data points
+    with your cursor on the frontend, get it as a Pandas dataframe in Python!
+
+    **Example.**
+
+    ```python
+    import altair as alt
+    import marimo as mo
+    from vega_datasets import data
+
+    chart = alt.Chart(data.cars()).mark_point().encode(
+        x='Horsepower',
+        y='Miles_per_Gallon',
+        color='Origin',
+    )
+
+    chart = mo.ui.chart(chart)
+    ```
+
+    ```
+    # View the chart and selected data as a dataframe
+    mo.hstack([chart, chart.value])
+    ```
+
+    **Attributes.**
+
+    - `value`: the data if the chart has any selections
+    - `selections`: the selection of the chart; this may be an interval along
+        the name of an axis or a selection of points
+    - `dataframe`: a Pandas dataframe of the unfiltered chart data
+
+    **Initialization Args.**
+
+    - `chart`: An `altair.Chart` or a Vega-lite spec
+    - `chart_selection`: optional selection type,
+        `"point"`, `"interval"`, or a bool; defaults to `True` which will
+        automatically detect the best selection type
+    - `field_selection`: optional list of fields (columns) for which to
+        enable selecton, `True` to enable selection for all legend items, or
+        `False` to disable selection entirely
+    - `label`: optional text label for the element
+    - `on_change`: optional callback to run when this element's value changes
+    """
+
+    name: Final[str] = "marimo-vega"
+
+    def __init__(
+        self,
+        figure: Union[str, altair.Chart, VegaSpec],
+        chart_selection: Union[
+            Literal["point"], Literal["interval"], bool
+        ] = True,
+        field_selection: Union[List[str], bool] = True,
+        *,
+        label: str = "",
+        on_change: Optional[Callable[[pd.DataFrame], None]] = None,
+    ) -> None:
+        vega_spec = _parse_spec(figure)
+        self.dataframe = _to_dataframe(vega_spec)
+
+        if label:
+            vega_spec["title"] = label
+
+        # Automatically add full width
+        if "width" not in vega_spec:
+            vega_spec["width"] = "container"
+        if "vconcat" in vega_spec:
+            for chart in vega_spec["vconcat"]:
+                if "width" not in chart:
+                    chart["width"] = "container"
+            # without autosize, vconcat will overflow
+            if "autosize" not in vega_spec:
+                vega_spec["autosize"] = "fit-x"
+
+        # Selection for binned charts is not yet implemented
+        if _has_binning(vega_spec):
+            sys.stderr.write(
+                "Binning + selection is not yet supported in "
+                "marimo.ui.chart.\n"
+                "If you'd this feature, please file an issue: "
+                "https://github.com/marimo-team/marimo/issues\n"
+                "In the meantime, use `pd.cut` to bin data in Python "
+                "before creating the chart.\n"
```

Comment:
If it's short, can you include an example how to do this? I've never seen that function. Googled and it wasn't apparent to me how to use it.

Reply:
we can just remove this